.circleci/config.yml

```diff
@@ -45,15 +45,67 @@ jobs:
       - copy-gradle-properties
       - android/restore-gradle-cache
       - run:
-          name: Build
-          command: ./gradlew --stacktrace assembleDebug assembleRelease
+          name: Bintray Upload
+          command: |
+            # Use a different versioning style for builds from PRs to allow
+            # developers to iterate faster.
+            #
+            # All those dev builds will be deleted once the PR is merged by
+            # https://github.com/Automattic/bintray-garbage-collector/
+            #
+            # If $PREFIX is not set, that is, if we're not on a PR but a merge
+            # commit on develop or trunk, the version will be the value from
+            # the source code.
+            branch=$CIRCLE_BRANCH
+            if [[ -n "$CIRCLE_PULL_REQUEST" ]]; then
+              PR_NUMBER=$(basename $CIRCLE_PULL_REQUEST)
+              # Making the assumption that $CIRCLE_SHA1 will always be
+              # available.
+              PREFIX="$PR_NUMBER-$CIRCLE_SHA1"
+            elif [[ "$branch" != "trunk" && "$branch" != "develop" ]]; then
+              # This happens on the first push of a new branch, when there
+              # isn't a PR open for it yet.
+              echo "Running on a feature branch with no open PR: skipping Bintray upload"
+              exit 0
```

Comment:
If a build runs on a branch with no PR, then we wouldn't be able to generate the `PR-SHA` version.

This can happen when you push a branch to the remote but don't open the PR straight afterwards.

If you then open a PR and run the build again, CircleCI is smart enough to realize that there is now a PR for that commit and provide the environment variable.

[Here's](https://app.circleci.com/pipelines/github/mokagio/WordPress-Utils-Android/32/workflows/3fdea926-919f-40f8-b17b-9e4bdc46f5ec/jobs/42) an example of this behavior in action, I pushed a branch to my fork and did not open a PR for it

![image](https://user-images.githubusercontent.com/1218433/99473663-20194d00-299f-11eb-9d1e-76d982232476.png)

As a future refinement, I think it would be good to cancel the job here, just to make it clearer to the developers that no artifact was published.